go/vt/vtgateproxy/discovery.go

```diff
@@ -82,7 +82,6 @@ type JSONGateResolverBuilder struct {
 	targets   map[string][]targetHost
 	resolvers []*JSONGateResolver
 
-	rand     *rand.Rand
```

Comment:
I think it's ok to just use the global rand in the rand package here. We actually introduced a bug when we added `GetTargets` because `rand.Rand` is not safe to use concurrently (the top level package funcs are safe)

Reply:
That's fine, but then we need to call `rand.Seed` during Init somewhere

Reply:
[rand.Seed was deprecated in go1.20](https://github.com/cilium/ebpf/issues/933), so if we want to rebase off a current branch, we might want to avoid it. If possible, I'd like to avoid using package level global instances like `rand. globalRandGenerator` because they can be poisoned by other packages. If concurrency is a problem, could we use a rand.Rand and a sync.Mutex?

Reply:
noted, rewritten to use a rand.Rand with a sync.Mutex